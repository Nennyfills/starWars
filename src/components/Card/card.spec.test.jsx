import React from 'react';
import renderer from 'react-test-renderer';

import { Card } from './index';

describe('Card Component', () => {
  it('should mount the card wrapper', () => {
    const wrapper = mount(<Card />);
    const component = renderer.create(
      <Card
        horizontal={false}
        name="will"
        birthYear={2011}
        gender="male"
        openModal={() => true}
        films="films"
        species="species"
        vehicles="vehicles"
        starShip="starShips"
        showEachFilm="starShips"
        showEachSpecies="showEachSpecies"
        showEachStarShips="showEachStarShips"
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(wrapper.find('.card')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
