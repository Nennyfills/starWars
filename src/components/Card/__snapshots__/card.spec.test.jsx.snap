// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should mount the card wrapper 1`] = `
<div
  className="card"
>
  <section
    className="card_vertical"
  >
    <article
      className="vertical_design"
    >
      <img
        alt="loading"
        src="https://helpx.adobe.com/content/dam/help/en/stock/how-to/visual-reverse-image-search/jcr_content/main-pars/image/visual-reverse-image-search-v2_intro.jpg"
      />
    </article>
    <div
      className="fonts_card"
    >
      <div
        className="fonts_detail"
      >
        <div
          className="fonts_name"
        >
          will
        </div>
        <div
          className="fonts_gender"
        >
          male
        </div>
        <div
          className="fonts_birth"
        >
          2011
          Unavailable
        </div>
      </div>
      <div
        className="fonts_view"
      >
        <input
          className="view_btn"
          disabled={false}
          onClick={[Function]}
          type="button"
          value="View"
        />
      </div>
    </div>
  </section>
</div>
`;

exports[`Card Component should mount the card wrapper 2`] = `
<Card
  birthYear=""
  birthYearDescription="Unavailable"
  films="Unavailable"
  gender="Unavailable"
  horizontal={false}
  name="Unavailable"
  openModal={[Function]}
  showEachFilm=""
  showEachSpecies=""
  showEachStarShips=""
  showEachVehicles=""
  species="Unavailable"
  starShips=""
  vehicles="Unavailable"
>
  <div
    className="card"
  >
    <section
      className="card_vertical"
    >
      <article
        className="vertical_design"
      >
        <img
          alt="loading"
          src="https://helpx.adobe.com/content/dam/help/en/stock/how-to/visual-reverse-image-search/jcr_content/main-pars/image/visual-reverse-image-search-v2_intro.jpg"
        />
      </article>
      <div
        className="fonts_card"
      >
        <div
          className="fonts_detail"
        >
          <div
            className="fonts_name"
          >
            Unavailable
          </div>
          <div
            className="fonts_gender"
          >
            Unavailable
          </div>
          <div
            className="fonts_birth"
          >
            Unavailable
          </div>
        </div>
        <div
          className="fonts_view"
        >
          <Button
            button="view_btn"
            disabled={false}
            handleClick={[Function]}
            name="View"
          >
            <input
              className="view_btn"
              disabled={false}
              onClick={[Function]}
              type="button"
              value="View"
            />
          </Button>
        </div>
      </div>
    </section>
  </div>
</Card>
`;
